src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -250,5 +250,69 @@ void IEnumerator.Reset()
 
             void IDisposable.Dispose() { }
         }
+
+        public bool TryGetValue(string key, out Group value)
+        {
+            if (ContainsKey(key))
+            {
+                value = this[key];
+                return true;
+            }
+            value = null;
+            return false;
+        }
+
+        public bool ContainsKey(string key)
+        {
+            return _match._regex.GroupNumberFromName(key) >= 0;
+        }
+
+        public IEnumerable<string> Keys
+        {
+            get
+            {
+                return _match._regex.GetGroupNames();
+            }
+        }
+
+        public IEnumerable<Group> Values
+        {
+            get
+            {
+                return this._groups;
+            }
+        }
+
+        private sealed class EnumeratorKeyValue : IEnumerator<KeyValuePair<string, Group>>
+        {
+            private Enumerator _collectionEnumerator;
+
+            public EnumeratorKeyValue(GroupCollection collection)
+            {
+                _collectionEnumerator = new Enumerator(collection);
+            }
+
+            public KeyValuePair<string, Group> Current
+            {
+                get
+                {
+                    var value = _collectionEnumerator.Current;
+
+                    return new KeyValuePair<string, Group>(value.Name, value);
+                }
+            }
+
+            bool IEnumerator.MoveNext() => _collectionEnumerator.MoveNext();
+
+            void IDisposable.Dispose() => ((IDisposable)_collectionEnumerator).Dispose();
+            void IEnumerator.Reset() => ((IEnumerator)_collectionEnumerator).Reset();
+            object IEnumerator.Current => Current;
+        }
+
+        IEnumerator<KeyValuePair<string, Group>> IEnumerable<KeyValuePair<string, Group>>.GetEnumerator()
```

Comment:
Nit: this should be moved up above the nested types

Reply:
fixed